src/client/gpm_common.c

```diff
@@ -289,8 +289,7 @@ static int gpm_epoll_wait(struct gpm_ctx *gpmctx, uint32_t event_flags)
     epoll_ret = epoll_ctl(gpmctx->epollfd, EPOLL_CTL_ADD, gpmctx->fd, &ev);
```

Comment:
Probably worth a comment describing our intentions here. We maintain a persistent `epollfd` and a `timerfd`, adding the `timerfd` to the `epollfd` at initialization (in `gpm_epoll_setup`). When we're looking to use `gpmctx->fd`, we add it to the epoll and wait for either `fd` to fire. The intent here is to return (as far up the stack as possible) when the timer expires if we've made no progress. But if we've made progress (i.e,. reading or writing to `timerfd`), we could reset the timer and continue. This is meant to be a "global" timeout, meaning we're at least as responsive as this timeout. (and continue on to the next request if this one failed).